terraform/modules/spack/opensearch.tf

```diff
@@ -23,17 +30,25 @@ resource "aws_opensearch_domain" "spack" {
     zone_awareness_enabled = true
   }
 
-  cognito_options {
-    enabled          = true
-    identity_pool_id = "us-east-1:ff2664d7-a403-42ba-8407-5d90b3eaa948" # TODO: encode this into terraform
-    role_arn         = aws_iam_role.opensearch_cognito_role.arn
-    user_pool_id     = "us-east-1_k6YnDTVBT" # TODO: encode this into terraform
+  # TODO: our AWS Cognito config for OpenSearch is not encoded in Terraform yet.
+  # We want the existing Cognito set up for our production OpenSearch cluster to
+  # remain in use, so we use a dynamic block to ensure it's ignored for non-production
+  # OpenSearch deployments.
+  dynamic "cognito_options" {
+    for_each = local.cognito_enabled ? [1] : []
+
+    content {
+      enabled          = true
+      identity_pool_id = "us-east-1:ff2664d7-a403-42ba-8407-5d90b3eaa948" # TODO: encode this into terraform
+      role_arn         = aws_iam_role.opensearch_cognito_role[0].arn
```

Comment:
Any reason this went from `aws_iam_role.opensearch_cognito_role.arn` to `aws_iam_role.opensearch_cognito_role[0].arn` in this change?

Reply:
The `aws_iam_role.opensearch_cognito_role` went from a string to an array due to the `count` argument (which we're using to create the resource conditionally)

https://developer.hashicorp.com/terraform/language/meta-arguments/count